invenio_circulation/permissions.py

```diff
@@ -25,7 +25,9 @@ def check_permission(permission):
 
     :param permission: The permission to check.
     """
-    if permission and not permission.can():
+    # NOTE: we have to explicitly check for not None, since flask-principal
+    # overrides the default __bool__ implementation for permission.
+    if permission is not None and not permission.can():
```

Comment:
Any related code/issue for that? Just out of curiosity!

Reply:
https://github.com/mattupstate/flask-principal/blob/0b5cd06b464ae8b60939857784bda86c03dda1eb/flask_principal.py#L227